### PR TITLE
Adding client to receive & send to subscribing clients

### PR DESCRIPTION
**Pull request by renukamanavalan**

### CODE REVIEW

File: gnmi_server/client_subscribe.go [hunk 1]
```diff
@@ -30,17 +30,34 @@ type Client struct {
...
 func NewClient(addr net.Addr) *Client {
 	pq := queue.NewPriorityQueue(1, false)
 	return &Client{
 		addr: addr,
 		q:    pq,
+		logLevel: logLevelError,
 	}
 }
 
+func (c *Client) setLogLevel(lvl int) {
```
Comment (zbud-msft) [inline gnmi_server/client_subscribe.go:51]: Inconsistent use of tabs/spaces

File: telemetry/telemetry.go [hunk 2]
```diff
@@ -60,6 +61,13 @@ func main() {
 	cfg.Port = int64(*port)
 	var opts []grpc.ServerOption
 
+	cfg.LogLevel, _ = strconv.Atoi(getflag("v"))
+	log.Errorf("flag: log level %v", cfg.LogLevel)
```
Comment (zbud-msft) [inline telemetry/telemetry.go]: Doesn't look like you are catching error on 64. You log error when you don't know if error occurred.

File: sonic_data_client/events_client.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+const MISSED_BUFFSZ = 16
+
+func NewEventClient(paths []*gnmipb.Path, prefix *gnmipb.Path, logLevel int) (Client, error) {
+    var evtc EventClient
+    evtc.prefix = prefix
+    for _, path := range paths {
+        // Only one path is expected. Take the last if many
+        evtc.path = path
+    }
+    log.Errorf("NewEventClient constructed. logLevel=%d", logLevel)
```
Comment (zbud-msft) [inline sonic_data_client/events_client.go]: Why do we use errorf, when no error has occured?

File: sonic_data_client/events_client.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    subs_data := make(map[string]interface{})
+    subs_data["recv_timeout"] = SUBSCRIBER_TIMEOUT
+    j, err := json.Marshal(subs_data)
+    if err != nil {
+        log.Errorf("events_init_subscriber: Failed to marshal")
+        return nil, err
+    }
+    js := string(j)
+    evtc.subs_handle = C.events_init_subscriber_wrap(C.CString(js))
+    log.Errorf("events_init_subscriber: h=%v", evtc.subs_handle)
```
Comment (zbud-msft) [inline sonic_data_client/events_client.go]: same

File: sonic_data_client/events_client.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+type EventClient struct {
+
+    prefix      *gnmipb.Path
+    path        *gnmipb.Path
+
+    q           *queue.PriorityQueue
+    channel     chan struct{}
+
+    w           *sync.WaitGroup // wait for all sub go routines to finish
```
Comment (zbud-msft) [inline sonic_data_client/events_client.go]: maybe we can use wg or waitgroup as variable name?

**Review by zbud-msft — COMMENTED** [review]
Also need to add sonic-mgmt tests

File: gnmi_server/client_subscribe.go (hunk 1 above)
Comment (renukamanavalan) [reply gnmi_server/client_subscribe.go:51]: switched to tabs

File: telemetry/telemetry.go (hunk 2 above)
Comment (renukamanavalan) [reply telemetry/telemetry.go]: V is part of flags and its value is restricted by flags package. Hence ignored.
But fixed now.

File: sonic_data_client/events_client.go (hunk 3 above)
Comment (renukamanavalan) [reply sonic_data_client/events_client.go]: initially used it like printf. I will change to debug

File: sonic_data_client/events_client.go (hunk 4 above)
Comment (renukamanavalan) [reply sonic_data_client/events_client.go]: resolved

File: sonic_data_client/events_client.go (hunk 5 above)
Comment (renukamanavalan) [reply sonic_data_client/events_client.go]: switched to wg

**Review by zbud-msft — APPROVED** [review]

**Pull request merged**